apps/meteor/client/views/marketplace/helpers.ts

```diff
@@ -175,9 +177,10 @@ export const appButtonProps = ({
 	versionIncompatible,
 	isAdminUser,
 	requestedEndUser,
-}: App & { isAdminUser: boolean }): appButtonResponseProps | undefined => {
+	endUserRequested,
```

Comment:
what is the difference between endUserRequested and requestedEndUser?

Reply:
requestedEndUser is a boolean, that comes from the marketplace API, which signals if the current app has been requested by the current user. endUserRequested is a local state variable used as a palliative to disable the request button immediately upon requesting the current app. It was implemented due to the AppProvider not properly refetching the list of apps after a request was made and the reload() function was called, and, consequently, the "new" requestedEndUser boolean was not being retrieved to disable the button "concretely".